### ISSUE TITLE
Why are the days in the sidebar in a different fromat from the ones in the right pane?

### ISSUE THREAD

**Issue by elad661** [abrt/gnome-abrt#91]

**Comment by rluzynski** [comment]
My proposition is:

1. Stop using `python-humanize`. We have `fancydate` function instead which is probably not as fancy as `python-humanize` (e.g., displays `"3 weeks ago"` instead of `"20 days ago"` and `"2 years ago"` instead of `"2 years 3 months ago"`) but I think it is fancy enough. We can make it more fancy if you think it's necessary. Besides that, we have it translated into 11 languages and we are ready to accept more translations. This change will make the dates in the left and right panel displayed in the same format.
1. Reword `"Detected"` to `"First occurrence"`; this term is widely used at https://retrace.fedoraproject.org. We'll be consistent with the FAF server and we'll avoid confusion why the dates are different.

What do you think?

**Comment by jfilak** [comment]
I agree that we need to sort out the issue with translations but I am not in a favor of re-developing functions to show human-readable dates. I added the `fancydate` function because I wanted to avoid the dependency bloat. However, I was forced to add dependency on `python-humanize` because I didn't want to re-implement it and I was asked to make the date format more fine grained.

> We can make it more fancy if you think it's necessary.

We should just fix it (e.g. do not show `Last year` for crashes detected in December in January )

> while python-humanize is currently translated into 4 languages only and the new languages are not so easy to add.

When @mhabrnal  approached `python-humanize` upstream with a request to translate it to more languages they asked him to do it himself but they weren't against it. How about to approach them with a humble offer to help them to set up an account at Zanata or Transifex?

> This change will make the dates in the left and right panel displayed in the same format.

I think the root of this issue is that the dates are different but users expect same values.

> Reword "Detected" to "First occurrence"

OK, It makes sense.

**Comment by allanday** [comment]
> Reword "Detected" to "First occurrence"
> OK, It makes sense.

Rafal asked me about this. "First Occurrence" seems like an improvement, but "occurrence" isn't a very common word in English. "First Detected" could be an alternative.

**Comment by rluzynski** [comment]
Thanks, Allan, this is clarified now.

Now @jfilak: as far as I understood:
* different date formats are intentional and correct,
* it is necessary and correct to maintain both `fancydate` and `python-humanize` at the moment,
* optionally `fancydate` will be dropped ever in future.

> > We can make it more fancy if you think it's necessary.
>
> We should just fix it (e.g. do not show `Last year` for crashes detected in December in January )

This is new and beyond the scope of this issue but sounds reasonable and probably simple enough to discuss it here. So you want to display:
* `Yesterday` even if yesterday was also a previous month or previous year, etc. (this works already);
* `Monday`, `Tuesday` etc. as long as it was less than 7 days ago, even if it was also a previous month/year; for example, today is Tuesday but we'll display `Wednesday` if the crash occurred 6 days ago (even if 6 days ago was a different month/year); note that in that approach `Last week` will be displayed only if the time difference is >= 7 days but we have not crossed more than 1 weekend;
* `Last month` if we are in January and it happened in December (even although it was also last year).

Is that correct and sufficient? Is it OK to refer to all weekdays from the previous week? Is it OK to refer to the `Last week` and `Last month` only or should we also refer to `2 months ago` (and maybe more, **how much more?**), for example if we are in January and it happened in November should we display `2 months ago` or `Last year`? I think that `Last week` and `Last month` instead of `Last year` is sufficient but I am not a designer.

> When @mhabrnal approached `python-humanize` upstream with a request to translate it to more languages they asked him to do it himself but they weren't against it. How about to approach them with a humble offer to help them to set up an account at Zanata or Transifex?

That's probably a good idea but I will not help them because first I must teach myself how to do it. :-) As a quick start you can deliver them those translations we already have here for `fancydate`.

**Comment by allanday** [comment]
My suggestion would be to only use the human readable version for recent events, and then switch to a more exact date for anything further in the past. Otherwise you will end up with lots of repeated "Last month" and "Last year" strings, which don't tell you very much. This is how Nautilus does it, for example.

Today: <hour>:<minute>
Yesterday: "Yesterday"
Yesterday - 7 days ago: <day of the week>
Anything after that, this year: <day number> <short month> (example: "2 Oct")
Anything after that, other years: <day number> <short month> <year> (example: "2 Oct 2015")

**Comment by rluzynski** [comment]
Thank you, @allanday, for your suggestion. My humble opinion is that this change makes sense but we should implement it only after F25 release as a preparation for the future F26 release. What do you think, @jfilak? What do other people think?

**Comment by jfilak** [comment]
@rluzynski It's good you think about Fedora, but we should stop development just because of  inappropriate Fedora life phase. We do branching of gnome-abrt for stable Fedora releases, so you can implement it, let it live in master and release it to Fedora with Fedora 26.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/abrt/gnome-abrt/issues/91#issuecomment-52298708

@elad661, that is now consistent:

<p><img width="685" height="719" alt="Image" src="https://github.com/user-attachments/assets/12d49060-096d-4707-91e9-b99bb2106b43" /></p>
